.github/workflows/tests.yml

```
@@ -24,91 +30,134 @@ jobs:
+          bash ci/install_geos.sh
 
       - name: Install python dependencies
-        shell: bash
         run: |
-          pip install --disable-pip-version-check --upgrade pip
-          pip install --upgrade wheel
-          pip install git+https://github.com/pygeos/pygeos.git -v
+          python -m pip install --disable-pip-version-check --upgrade pip
+          pip install --upgrade wheel setuptools
           if [ "$GEOS_VERSION" = "main" ]; then
-            pip install --upgrade --pre Cython numpy;
-          elif [ -n "${{ matrix.numpy }}" ]; then
-            pip install --upgrade numpy==${{ matrix.numpy }};
+            pip install --upgrade --pre Cython numpy pytest pytest-cov coveralls;
           else
-            pip uninstall --yes numpy;
+            pip install --upgrade Cython numpy==${{ matrix.numpy }} pytest pytest-cov coveralls;
           fi
-          pip install --upgrade coveralls pytest-cov pytest
           pip list
 
-      - name: Set environment variables
-        shell: bash
+      - name: Set environment variables (Linux)
+        run: |
+          echo "${{ env.GEOS_INSTALL }}/bin" >> $GITHUB_PATH
+          echo "LD_LIBRARY_PATH=${{ env.GEOS_INSTALL }}/lib" >> $GITHUB_ENV
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+
+      - name: Set environment variables (OSX)
+        run: |
+          echo "${{ env.GEOS_INSTALL }}/bin" >> $GITHUB_PATH
+          echo "LDFLAGS=-Wl,-rpath,${{ env.GEOS_INSTALL }}/lib" >> $GITHUB_ENV
+        if: ${{ matrix.os == 'macos-latest' }}
+
+      # Windows requires special treatment:
+      # - geos-config does not exist, so we specify include and library paths
+      # - Python >=3.8 ignores the PATH for finding DLLs, so we copy them into the package 
+      - name: Set environment variables + copy DLLs (Windows)
         run: |
-          echo "LD_LIBRARY_PATH=$GEOS_INSTALL/lib" >> $GITHUB_ENV
-          echo $GEOS_INSTALL/bin >> $GITHUB_PATH
+          cp geosinstall/geos-${{ matrix.geos }}/bin/*.dll shapely
+          echo 'GEOS_LIBRARY_PATH=${{ env.GEOS_INSTALL }}\lib' >> $GITHUB_ENV
+          echo 'GEOS_INCLUDE_PATH=${{ env.GEOS_INSTALL }}\include' >> $GITHUB_ENV
+        if: ${{ matrix.os == 'windows-2019' }}
 
       - name: Build and install Shapely
-        shell: bash
         run: |
-          pip install -v -e .
+          python setup.py build_ext --inplace
+          pip install . --no-build-isolation
 
       - name: Overview of the Python environment (pip list)
-        shell: bash
-        run: |
-          pip list
+        run: pip list
 
       - name: Run tests
-        shell: bash
         continue-on-error: ${{ matrix.geos == 'main' }}
         run: |
-          python -c "from shapely import geos; print(geos.geos_version_string)"
-          python -m pytest --cov shapely --cov-report term-missing
+          pytest shapely/tests -cov shapely --cov-report term-missing ${{ matrix.extra_pytest_args }}
```

```suggestion
          pytest tests shapely/tests -cov shapely --cov-report term-missing ${{ matrix.extra_pytest_args }}
```
And drop the next line.